src/server/mod.rs

```diff
@@ -412,17 +410,24 @@ impl<S, B> Server<S, B>
             let addr = socket.remote_addr;
             debug!("accepted new connection ({})", addr);
 
-            let service = new_service.new_service()?;
+            let service = match new_service.new_service() {
+                Ok(service) => service,
+                Err(err) => return future::Either::Left(future::err(err))
+            };
+            
             let s = NotifyService {
                 inner: service,
                 info: Arc::downgrade(&info_cloned),
             };
             info_cloned.lock().unwrap().active += 1;
-            let fut = protocol.serve_connection(socket, s)
-                .map(|_| ())
-                .map_err(move |err| error!("server connection error: ({}) {}", addr, err));
-            spawn(fut);
-            Ok(())
+            let fut = protocol.serve_connection(socket, s).then(move |result| {
```

Comment:
I think this can be `FutureExt::recover` also.

Reply:
Will do!